### ISSUE TITLE
Issues with MEME and FEL analysis

### ISSUE THREAD

**Issue by Pluto2013** [veg/hyphy#709]
Using the latest version of online datamonkey webportal to re-run MEME and FEL analysis with previous dataset. Below is the details of the sequence alignment file.
- 370 total sequences
- size: 10176 nt (3392 AA)
The runs keep having the same error messages as below:
"Could not contact server for job status updates" 
and received email notification with titled "Something went wrong with your job: 5a1cbc356708bb37f04cc081"
It seems that the analysis stops half way before completing the run.
I never had such issues with the previous online version using the same dataset done sometimes ago (early this year). 
Your helps will be greatly appreciated.

**Comment by stevenweaver** [comment]
Dear @Pluto2013,

Apologies for the delayed response. Can you try resubmitting the job? 

We had an unexpected issue with logging earlier this week, and in combination with a holiday and an unexpectedly busy week back, it took longer than usual to fully resolve the issue. Operations should return to normal today.  

Best,
Steven

**Comment by Pluto2013** [comment]
Hi Steven, thanks for your responses on the issues reported.
Tried to re-submit the job two days ago, again the analyses stopped at about 50%.
For FEL analysis, total number of codon is 3392 for our dataset but the run stopped at about 1800 and no further update on the result after 24 hours (refresh the webpage). Any idea on the duration per run based on the details of dataset provided previously? Your advise will be greatly appreciated.

**Comment by Pluto2013** [comment]
Hi Steven, the FEL analysis is successful. 
However, the MEME analysis keep having the error messages and stop at 30 to 40% run.
FYI please. Is there any limitation on the number/length of DNA sequences for the job submission?

**Comment by stevenweaver** [comment]
Dear @Pluto2013 , 

I'm investigating now. I should have an explanation shortly. 
The maximum dataset for MEME and FEL is 500 sequences and 12000 sites.

Best,
Steven

**Comment by stevenweaver** [comment]
Dear @Pluto2013, 

I've identified your job and have confirmed the issues you are experiencing. I've started running your meme job with our fastest compute node and am monitoring the process closely.

The job was submitted on December 2nd and failed around 11 hours into the analysis. My email is sweaver@temple.edu. If you email me I can send the results to you privately. 

Longer running jobs that fail like this are the ones I care most about, and I will keep this issue open until I confirm my suspicions as to what went wrong and push a patch fix. 

Best,
Steven

**Comment by stevenweaver** [comment]
Update : This dataset causes failures at non-deterministic points within the HyPhy process itself. Will need to investigate the dataset.

**Comment by spond** [comment]
Dear @stevenweaver,

Can you run this alignment with the current `beta` branch? Perhaps my fixes to issue #703 have resolved it?

Best,
Sergei

**Comment by Pluto2013** [comment]
Hi Steven,

Thanks for following up the issue. 
Please note that I have sent a request to your email.
Hope to receive the results soon.
Many thanks in advance!

**Comment by stevenweaver** [comment]
We are still experiencing an issue with this dataset in HyPhy 2.3.8. 

The stack trace 

```
Function call stack
1 : Optimize storing into, results, the following likelihood function:^lf_bsrel ;
-------
2 : Call(result_callback,0,Eval(job+'('+Join(",",utility.Map(arguments,"_value_","utility.convertToArgumentString (_value_)"))+')'),arguments)
-------
3 : mpi.QueueJob(meme.queue,"meme.handle_a_site",{"0":"meme.site_likelihood","1":"meme.site_likelihood_bsrel","2":alignments.serialize_site_filter((meme.filter_specification[meme.partition_index])[utility.getGlobalValue("terms.data.name")],(_pattern_info_[utility.getGlobalValue("terms.data.sites")])[0]),"3":meme.partition_index,"4":_pattern_info_,"5":meme.site_model_mapping},"meme.store_results")
-------
4 : ExecuteCommands in string transform using basepath .
-------
5 : utility.ForEachPair(meme.site_patterns,"_pattern_","_pattern_info_",'
            if (_pattern_info_[utility.getGlobalValue("terms.data.is_constant")]) {
                meme.store_results (-1,None,{"0" : "meme.site_likelihood",
                                                                         "1" : "meme.site_likelihood_bsrel",
                                                                                         "2" : None,
                                                                                         "3" : meme.partition_index,
                                                                                         "4" : _pattern_info_,
                                                                                         "5" : meme.site_model_mapping
                                     });
            } else {
                mpi.QueueJob (meme.queue, "meme.handle_a_site", {"0" : "meme.site_likelihood",
                                                                                                                 "1" : "meme.site_likelihood_bsrel",
                                                                 "2" : alignments.serialize_site_filter
                                                                   ((meme.filter_specification[meme.partition_index])[utility.getGlobalValue("terms.data.name")],
                                                                   (_pattern_info_[utility.getGlobalValue("terms.data.sites")])[0]),
                                                                                                                                 "3" : meme.partition_index,
                                                                                                                                 "4" : _pattern_info_,
                                                                                                                                 "5" : meme.site_model_mapping
                                                                                                                                        },
                                                                                                                                        "meme.store_results");
            }
        ')
-------
Error:
HyPhy killed by signal 15Error:
HyPhy killed by signal 15
```

and from error.log:
```
Error:Both arguments must be Strings (or a String Literal and a number) in an iterator call for Associative Arrays
```

**Comment by spond** [comment]
Dear @Pluto2013 and @stevenweaver,

This error is different from #703. I will investigate shortly. @stevenweaver, can you please share the file with me by e-mail?

Best,
Sergei

**Comment by stevenweaver** [comment]
Dear @spond, 

I've created a directory with the dataset in question on our computing cluster and emailed you the directory path. 

Best,
Steven

**Comment by spond** [comment]
Dear @stevenweaver,

Received, debugging. May take a while due to the large file size which implies long wait times to hit error conditions.

Best,
Sergei

**Comment by spond** [comment]
Dear @stevenweaver,

Should be resolved with commit 621da1d525f91791e262082cd2dd1762eb8f4523

Best,
Sergei
